test/e2e/encryption_at_rest_test.go

```diff
@@ -294,3 +302,223 @@ var _ = Describe("Encryption at rest AWS", Label("encryption-at-rest"), func() {
+	project := &v1.AtlasProject{}
+	err := k8sClient.Get(ctx, client.ObjectKey{Namespace: ns, Name: name}, project)
+	if err != nil {
+		return nil, err
+	}
+	for _, condition := range project.Status.Conditions {
+		if condition.Type == statusType {
+			return &condition, nil
+		}
+	}
+	return nil, fmt.Errorf("condition %s not found. found %v", statusType, project.Status.Conditions)
+}
+
+func createProjectWithValidationError(testData *model.TestDataProvider, errMsg string) {
+	if testData.Project.GetNamespace() == "" {
+		testData.Project.Namespace = testData.Resources.Namespace
+	}
+	By(fmt.Sprintf("Deploy Broken Project %s", testData.Project.GetName()), func() {
+		err := testData.K8SClient.Create(testData.Context, testData.Project)
+		Expect(err).ShouldNot(HaveOccurred(), "Project %s was not created", testData.Project.GetName())
+		Eventually(func(g Gomega) {
+			condition, err := getProjectCondition(
+				testData.Context,
+				testData.K8SClient,
+				status.ValidationSucceeded,
+				testData.Resources.Namespace,
+				testData.Project.GetName(),
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+			g.Expect(condition).ToNot(BeNil())
+		}).WithPolling(5 * time.Second).WithTimeout(40 * time.Second).Should(Succeed())
+		condition, err := getProjectCondition(
+			testData.Context,
+			testData.K8SClient,
+			status.ValidationSucceeded,
+			testData.Resources.Namespace,
+			testData.Project.GetName(),
+		)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(string(condition.Status)).To(Equal("False"))
+		Expect(condition.Message).To(ContainSubstring(errMsg))
+	})
+}
+
+func withProperUrls(properties string) string {
+	urls := `"auth_uri": "https://accounts.google.com/o/oauth2/auth",
+	"token_uri": "https://oauth2.googleapis.com/token",
+	"auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+	"client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/619108922856-compute%40developer.gserviceaccount.com"`
+	return fmt.Sprintf(`{%s, %s}`, urls, properties)
+}
+
+func repeat(unit string, times int) string {
+	var buf strings.Builder
+	for i := 0; i < times; i++ {
+		buf.WriteString(unit)
+	}
+	return buf.String()
+}
+
+func yamlMultiline(indentation int, s string) string {
+	indentPrefix := repeat(" ", indentation)
+	var buf strings.Builder
+	fmt.Fprintf(&buf, "|\n")
+	scanner := bufio.NewScanner(bytes.NewBufferString(s))
+	for scanner.Scan() {
+		line := scanner.Text()
+		fmt.Fprintf(&buf, "%s%s\n", indentPrefix, line)
+	}
+	if err := scanner.Err(); err != nil {
+		return err.Error()
+	}
+	return buf.String()
+}
+
+const projectWithGceEncryptionFmt = `apiVersion: atlas.mongodb.com/v1
+kind: AtlasProject
+metadata:
+  name: my-project
+spec:
+  name: Test Atlas Operator Project
+  encryptionAtRest:
+    googleCloudKms:
+      enabled: true
+      keyVersionResourceID: %s
+      serviceAccountKey: %s`
+
+// composeGoogleEncryptionAtRestProjectYAML produces something like this YAML:
```

Comment:
Should we remove those commented block?

Reply:
I would leave it, I think it is helpful to quickly visualize what the code is trying to output and this functions is unlikely to change over time.